### ISSUE TITLE
Using using with NetMQ 

### ISSUE THREAD

**Issue by drewnoakes** [zeromq/netmq#263]
Ok, that's a fairly cryptic title :)

Throughout working with NetMQ I've found that code is easier to read when it's less nested. I'd like to know whether there are any objections to this idea, before I go ahead an apply it throughout the sample code.

Here's an example:

### Before

```csharp
using (NetMQContext ctx = NetMQContext.Create())
{
    using (var server = ctx.CreateResponseSocket())
    {
        server.Bind("tcp://127.0.0.1:5556");
        using (var client = ctx.CreateRequestSocket())
        {
            client.Connect("tcp://127.0.0.1:5556");

            client.Send("Hello");
            server.ReceiveString();
        }
    }
}
```

### After

```csharp
using (var context = NetMQContext.Create())
using (var server = context.CreateResponseSocket())
using (var client = context.CreateRequestSocket())
{
    server.Bind("tcp://localhost:5556");
    client.Connect("tcp://localhost:5556");

    client.Send("Hello");
    server.ReceiveString();
}
```

My theory is that the second pattern is easier to read, which should help newcomers to NetMQ and ZeroMQ see the underlying patterns in sample code.

I've already several of the samples in the docs (`*.md`) to this format in order to make them fit within the page's width.

But before I try and apply this to the sample project's (`*.cs`) I'd like to give others a chance to object.

**Comment by somdoron** [comment]
I think it's great idea.

**Comment by sachabarber** [comment]
Looks good to me too

**Comment by drewnoakes** [comment]
Closed via #268.
